### PR TITLE
Remove custom exceptions

### PR DESCRIPTION
**Pull request by phako**

### CODE REVIEW

File: tests/hardware/src/forwardBackwardDFMC-MD22.cpp [hunk 1]
```diff
@@ -158,8 +158,11 @@ int main(int argc, char* argv[]) {
           CHECK_EXIT_ON_SIGNAL
         }
       }
-      catch(mtca4u::MotorDriverException& e) {
-        std::cout << "Problem reading target position" << std::endl;
+      catch(ChimeraTK::logic_error& e) {
```
Comment (toseko) [inline tests/hardware/src/forwardBackwardDFMC-MD22.cpp:161]: Why the logic_error may be thrown in this place at all. It looks like only runtime error should be possible here.

File: tests/hardware/src/forwardBackwardDFMC-MD22.cpp [hunk 2]
```diff
@@ -181,8 +184,11 @@ int main(int argc, char* argv[]) {
...
```
Comment (toseko) [inline tests/hardware/src/forwardBackwardDFMC-MD22.cpp:187]: And the same as for line 161. Should one see here the logic_error ?

File: tests/hardware/src/forwardBackwardDFMC-MD22.cpp (hunk 1 above)
Comment (phako) [reply tests/hardware/src/forwardBackwardDFMC-MD22.cpp:161]: Ah yes, that is most likely only a runtime error

File: backend/src/SPIviaPCIe.cc [hunk 3]
```diff
@@ -118,7 +116,7 @@ namespace mtca4u {
       if(errno == EFAULT) {
         std::stringstream errorMessage;
         errorMessage << "Error sleeping " << microSeconds << " micro seconds!";
-        throw(MotorDriverException(errorMessage.str(), MotorDriverException::SPI_TIMEOUT));
+        throw ChimeraTK::runtime_error(errorMessage.str());
```
[the remark this answers is not in the document]
Comment (phako) [reply backend/src/SPIviaPCIe.cc:119]: Right, good point

[the remark this answers is not in the document]
Comment (phako) [reply backend/src/SPIviaPCIe.cc:119]: Actually that should probably use something different than nanosleep as well, I will make a new ticket for this

**Pull request merged**